web/api/v1/api.go

```diff
@@ -568,10 +578,10 @@ func (api *API) series(r *http.Request) apiFuncResult {
 		metrics = append(metrics, set.At().Labels())
 	}
 	if set.Err() != nil {
-		return apiFuncResult{nil, &apiError{errorExec, set.Err()}, warnings, nil}
+		return apiFuncResult{nil, &apiError{errorExec, set.Err()}, warnings, closer}
 	}
 
-	return apiFuncResult{metrics, nil, warnings, nil}
+	return apiFuncResult{metrics, nil, warnings, closer}
```

Comment:
```suggestion
	return apiFuncResult{metrics, nil, warnings, q.Close}
```

Reply:
Computer says no.

Or more precisely: “cannot use q.Close (type func() error) as type func() in field value”

Reply:
ooh ok.